Week2/part1.js

```diff
@@ -0,0 +1,81 @@
+var mysql = require('mysql');
+var connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'new_world',
+});
+
+connection.connect();
+
+function sendQuery(query) {
+  connection.query(query, function(error, results, fields) {
+    if (error) {
+      throw error;
+    }
+    if (!results[0]) {
+      console.log('Sorry no match in our database!');
+    }
+    if (results.length === 0) {
+      console.log('False');
+    } else {
+      console.log('The reply is', results);
+    }
+  });
+}
+
+const inputType = process.argv[2]; // type like region, language, country
+const userInput = process.argv[3]; // value like western europe, Dutch, Netherlands
+const userInput2 = process.argv[4];
+
+//What is the capital of country X ? (Accept X from user)
+const capital_of_country = `SELECT capital FROM country WHERE country.Name = '${userInput}'`;
+
+// List all the languages spoken in the region Y (Accept Y from user)
+const language_of_region = `SELECT Language FROM countrylanguage INNER JOIN country ON country.Code = countrylanguage.CountryCode WHERE country.Region = '${userInput}'`;
+
+function help() {
+  console.log(`
+  - to find the capital of country X enter 'country' followed by the name of the country you want.
+  - to List all the languages spoken in the region Y enter 'region' followed by the region you want.
+  - to find the number of cities in which language Z is spoken enter 'language' followed by the language you want.
+`);
+}
+
+//Find the number of cities in which language Z is spoken (Accept Z from user)
+const language_of_city = `SELECT COUNT(Name) AS CityLanguage FROM city INNER JOIN countrylanguage ON countrylanguage.CountryCode = city.Country_Code WHERE countrylanguage.Language = '${userInput}'`;
+
+//Accept the region and language from the user. Are there any countries in this region with the given language as the official language ? If yes, display those countries. If no, display FALSE. E.g. (A) input region : 'Western Europe' and input language : 'Dutch' output should be Belgium and Netherlands (B) input region : 'Western Europe' and input language : 'Hindi' output should be 'FALSE'
+
+const official_language = `SELECT Name FROM country INNER JOIN countrylanguage ON countrylanguage.CountryCode = country.Code WHERE countrylanguage.Language = '${userInput}' AND country.Region = '${userInput2}' AND countrylanguage.isOfficial ='T'`;
+
+// List all the continents with the number of languages spoken in each continent
+const language_of_continents =
+  'SELECT COUNT(countrylanguage.Language) as languages, country.continent FROM country INNER JOIN countrylanguage ON countrylanguage.CountryCode = country.Code GROUP BY country.continent';
```

Comment:
Suppose there is a small continent with two countries that speak the same language. On this continent only 1 language is spoken. However your query would return a wrong result. How can this be fixed?